tools/compile-keymap.c

```diff
@@ -114,29 +138,59 @@ parse_options(int argc, char **argv, struct xkb_rule_names *names)
+        {"Wunrecognized-keysym",                  no_argument, &warnings, XKB_WARNING_UNRECOGNIZED_KEYSYM},
+        {"Wno-unrecognized-keysym",               no_argument, &warnings, -XKB_WARNING_UNRECOGNIZED_KEYSYM},
+        {"Wnumeric-keysym" ,                      no_argument, &warnings, XKB_WARNING_NUMERIC_KEYSYM},
+        {"Wno-numeric-keysym",                    no_argument, &warnings, -XKB_WARNING_NUMERIC_KEYSYM},
+        {"Wconflicting-key-type",                 no_argument, &warnings, XKB_WARNING_CONFLICTING_KEY_TYPE},
+        {"Wno-conflicting-key-type",              no_argument, &warnings, -XKB_WARNING_CONFLICTING_KEY_TYPE},
+        {"Wconflicting-key-action",               no_argument, &warnings, XKB_WARNING_CONFLICTING_KEY_ACTION},
+        {"Wno-conflicting-key-action",            no_argument, &warnings, -XKB_WARNING_CONFLICTING_KEY_ACTION},
+        {"Wconflicting-key-keysym",               no_argument, &warnings, XKB_WARNING_CONFLICTING_KEY_KEYSYM},
+        {"Wno-conflicting-key-keysym",            no_argument, &warnings, -XKB_WARNING_CONFLICTING_KEY_KEYSYM},
+        {"Wconflicting-key-symbol-map-fields",    no_argument, &warnings, XKB_WARNING_CONFLICTING_KEY_SYMBOL_MAP_FIELDS},
+        {"Wno-conflicting-key-symbol-map-fields", no_argument, &warnings, -XKB_WARNING_CONFLICTING_KEY_SYMBOL_MAP_FIELDS},
+        {"Wconflicting-modmap",                   no_argument, &warnings, XKB_WARNING_CONFLICTING_MODMAP},
+        {"Wno-conflicting-modmap",                no_argument, &warnings, -XKB_WARNING_CONFLICTING_MODMAP},
+        {"Wgroup-name-for-non-base-group",        no_argument, &warnings, XKB_WARNING_GROUP_NAME_FOR_NON_BASE_GROUP},
+        {"Wno-group-name-for-non-base-group",     no_argument, &warnings, -XKB_WARNING_GROUP_NAME_FOR_NON_BASE_GROUP},
+        {"Wmultiple-group-at-once",               no_argument, &warnings, XKB_WARNING_MULTIPLE_GROUP_AT_ONCE},
+        {"Wno-multiple-group-at-once",            no_argument, &warnings, -XKB_WARNING_MULTIPLE_GROUP_AT_ONCE},
+        {"Wcannot-infer-key-type",                no_argument, &warnings, XKB_WARNING_CANNOT_INFER_KEY_TYPE},
+        {"Wno-cannot-infer-key-type",             no_argument, &warnings, -XKB_WARNING_CANNOT_INFER_KEY_TYPE},
+        {"Wundefined-key-type",                   no_argument, &warnings, XKB_WARNING_UNDEFINED_KEY_TYPE},
+        {"Wno-undefined-key-type",                no_argument, &warnings, -XKB_WARNING_UNDEFINED_KEY_TYPE},
+        {"Wunresolved-keymap-symbol",             no_argument, &warnings, XKB_WARNING_UNRESOLVED_KEYMAP_SYMBOL},
+        {"Wno-unresolved-keymap-symbol",          no_argument, &warnings, -XKB_WARNING_UNRESOLVED_KEYMAP_SYMBOL},
+        {"Wall",                                  no_argument, &warnings, XKB_WARNING_ALL},
+        {"Wnone",                                 no_argument, &warnings, -XKB_WARNING_ALL},
+        {"Werror",                                no_argument, &warnings, XKB_WARNING_IS_ERROR},
```

Comment:
`Werror` is a large hammer, I personally prefer what e.g. `flake8` does where there's `W` and `E` for each warning. I can imagine something like `-Wconflicting-modmap -Eunrecognized-keysym`, it'd be nice if we could control it this way.

Reply:
Yes, still toying with it. I think I just have to add an `error_flags` field similar to the new `warning_flags`.

This would only affect the logging part atm though. I am not sure if it is easy to change the exit status to reflect those errors. The exit status could be a mask of the errors flags.

Reply:
I implemented the `error_flags` idea.